rest-api/rest-api.php

```diff
@@ -0,0 +1,672 @@
+    }
+
+    public function contacts( WP_REST_Request $request ) {
+
+        dt_write_log( 'Contacts' );
+
+        $fields     = $request->get_json_params() ?? $request->get_body_params();
+
+        //Converts Maarifa field names to DT field names
+        $fields2 = $this->map_fields_to_contact( $fields );
+
+        dt_write_log( $fields2 );
+
+        $url_params = $request->get_url_params();
+        $get_params = $request->get_query_params();
+        $silent     = isset( $get_params['silent'] ) && $get_params['silent'] === 'true';
+
+    //                    $check_dups = ! empty( $get_params['check_for_duplicates'] ) ? explode( ',', $get_params['check_for_duplicates'] ) :
+        $check_dups = true; //TO DO
+
+
+        $maarifa_contact_id = null;
+
+        if ( isset( $fields2['maarifa_data'] ) )
+        {
+
+            $maarifa_contact_id = $fields2['maarifa_data']['default']['id'];
+
+
+            if ( $maarifa_contact_id ) {
+
+                //Verify if maarifa_data already exists
+                $return_maarifa = $this->check_field_value_exists( $request, $maarifa_contact_id );
+
+            }
+
+
+            if ( isset( $return_maarifa ) && ! $return_maarifa == 0 )
+            {//Contact already exits in Maarifa
+
+
+                //Update the contact
+                $post_id_return = $return_maarifa[0]->post_id;
+
+                $this->update_maarifa_contact( $request, $post_id_return, $fields2 );
+
+            }
+            else //Contact is new
+            {
+
+                $post       = DT_Posts::create_post( 'contacts', $fields2, $silent, true, [
+                    'check_for_duplicates' => $check_dups,
+
+                ] );
+
+                dt_write_log( '1 post' );
+                dt_write_log( $fields );
+
+                // Country --> Locations
+                if ( !empty( $fields['street'] ) && !empty( $post['ID'] ) ) {
+
+                    dt_write_log( '$request' );
+                    dt_write_log( $request );
+
+                    $geoloc = $this->add_user_location( $request, $post['ID'] );
+                    $post['maarifa_data']['location_details'] = $geoloc;
+
+
+                    dt_write_log( 'geoloc' );
+                    dt_write_log( $geoloc );
+
+                    dt_write_log( 'post' );
+                    dt_write_log( $post );
+
+                }
+
+                return $post;
+            }
+        }
+
+        return null;
+    }
+
+
+    public function update_post( WP_REST_Request $request ){
+
+        $url_params = $request->get_url_params();
+        $id_upd = $url_params['id'];
+
+        $fields_orig     = $request->get_json_params() ?? $request->get_body_params();
+
+        //Converts Maarifa field names to DT field names
+        $fields3 = $this->map_fields_to_contact( $fields_orig );
+
+
+        $this->update_maarifa_contact( $request, $id_upd, $fields3 );
+    }
```

Comment:
Right now, this doesn't return anything in the JSON body of the response. The create endpoint returns the post, so this should return the same thing.